src/client/sandbox/child-window/index.ts

```diff
@@ -88,10 +88,44 @@ export default class ChildWindowSandbox extends SandboxBase {
 
             e.preventDefault();
 
-            this._openUrlInNewWindow(url);
+            this._openNewWindowAfterAllEventHandlers(url, e);
         });
     }
 
+    private _openNewWindowAfterAllEventHandlers (url, e) {
+        const nativeAddEventListener    = nativeMethods.windowAddEventListener || nativeMethods.addEventListener;
```

Comment:
Add the `NativeMethods.getAddEventListener`, `NativeMethods.getRemoveEventListener` methods and use them here and replace the 
`nativeMethods.windowAddEventListener || nativeMethods.addEventListener`, 
`nativeMethods.windowRemoveEventListener || nativeMethods.removeEventListener`
expressions across the whole module code.

Reply:
we already have nice methods for this:
https://github.com/DevExpress/testcafe-hammerhead/blob/master/src/client/sandbox/event/listeners.ts#L43